templates/cicd/pipeline_cfn.yml

```diff
@@ -235,27 +242,40 @@ Resources:
       Stages:
         - Name: Source
           Actions:
-            - Name: SourceCodeFor-{{$.AppName}}
-              ActionTypeId:
-                Category: Source
-                # TODO: #273 For CodeCommit, this needs to be AWS. Let's do that later
-                Owner: ThirdParty
-                Version: 1
-                Provider: {{.Source.ProviderName}}
-              Configuration:
-                # TODO: #273 For now this set of configurations is GitHub-specific,
-                # change it to be more generic
-                Owner: {{$.Source.Owner}}
-                Repo: {{$.Source.Repository}}
-                Branch: {{index $.Source.Properties "branch"}}
-                # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html#dynamic-references-secretsmanager
-                # Use the *entire* SecretString with version AWSCURRENT
-                OAuthToken: !Sub
-                    - '{{"{{"}}resolve:secretsmanager:${SecretId}{{"}}"}}'
-                    - { SecretId: !Ref GitHubAccessTokenSecretId }
-              OutputArtifacts:
-                - Name: SCCheckoutArtifact
-              RunOrder: 1
+            !If
+              - IsGitHub
+              -
+                - Name: SourceCodeFor-{{$.AppName}}
+                  ActionTypeId:
+                    Category: Source
+                    Owner: ThirdParty
+                    Version: 1
+                    Provider: {{.Source.ProviderName}}
+                  Configuration:
+                    Owner: {{$.Source.Owner}}
+                    Repo: {{$.Source.Repository}}
+                    Branch: {{index $.Source.Properties "branch"}}
+                    # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html#dynamic-references-secretsmanager
+                    # Use the *entire* SecretString with version AWSCURRENT
+                    OAuthToken: !Sub
+                      - '{{"{{"}}resolve:secretsmanager:${SecretId}{{"}}"}}'
+                      - { SecretId: {{$.Source.GitHubPersonalAccessTokenSecretID}} }
+                  OutputArtifacts:
+                    - Name: SCCheckoutArtifact
+                  RunOrder: 1
+              -
+                - Name: SourceCodeFor-{{$.AppName}}
+                  ActionTypeId:
+                    Category: Source
+                    Owner: AWS
+                    Version: 1
+                    Provider: {{.Source.ProviderName}}
+                  Configuration:
+                    RepositoryName: {{$.Source.Repository}}
+                    BranchName: {{index $.Source.Properties "branch"}}
+                  OutputArtifacts:
+                    - Name: CODE_ZIP
```

Comment:
dumb question: can the name of this output artifact also be `SCCheckoutArtifact` so that we don't have the conditional below?

Reply:
Smart question! You're right-- name is different from format, so both can be `SCCheckoutArtifact`. Thanks for catching that. And the default format is CODE_ZIP, so I'll just remove it. The full clone format is new and (I think) overkill for our purposes.